quipucords/api/source/serializer.py

```diff
@@ -363,7 +334,8 @@ def validate_name(name):
     @staticmethod
     def validate_hosts(hosts):
         """Make sure the hosts list is present."""
-        if not hosts:
+        hosts_list = json.loads(hosts)
+        if not hosts_list:
             raise ValidationError(_(messages.SOURCE_HOSTS_CANNOT_BE_EMPTY))
```

Comment:
json format validation could occur here, as I believe this will be called before the `create` method.

Reply:
Done.